Automattic/jetpack
# Protect: Fix ThreatFixHeader component title

## Fix
- The `ThreatFixHeader` component that displays for each threat in the `FixAllThreatsModal` is missing the title element.
- Unselecting all threats from the list in the modal, then reselecting them causes console errors and results in the threat IDs not being added back to the `threatIds` array to be fixed when initiated.

## Proposed changes:

* Update the modals that facilitate this component to ensure that the property containing the title is accessible and used
* Disable the `Fix all threats` button in the modal when all threats are unselected
* Fix the `handleCheckboxClick` functionality to avoid console errors

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1271

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Create a new Jurassic Ninja site and enable this branch of Protect using the Beta Tester plugin
* Add an upgrade and multiple fixable threats and run a scan
* Once located, connect your user and enter your credentials and click `Fix all threat` or within an individual threat accordion item `Fix threat`
* Verify that the threat title displays in either modal for each threat present
* From the `Fix all threats` modal ensure that you can unselect all threats, that the `Fix all` button becomes disabled, and that you can reselect threats (without console errors) and the `Fix all` buttons becomes available once again
* Ensure no regressions are introduced


## Screenshots:

Single threat:
![Screen Shot on 2024-07-11 at 14-46-47](https://github.com/user-attachments/assets/9cb92853-792a-4e39-ad22-8daae50a51c4)

Before w/ multiple threats:
![Screen Shot on 2024-07-11 at 13-48-29](https://github.com/Automattic/jetpack/assets/43220201/79117e89-8870-4a52-bcd8-294989aefc75)

After w/ multiple threats:
![Screen Shot on 2024-07-11 at 14-27-31](https://github.com/user-attachments/assets/4c8cdde3-f166-49dc-9cd1-a263bfc4d9ab)
